cli/pcluster/resources/batch/docker/scripts/generate_hostfile.sh

```diff
@@ -142,15 +142,15 @@ main() {
     check_arguments_valid
     check_batch_env_variables_exist
 
-    shared_dir_temp="${shared_dir}/${AWS_BATCH_JOB_ID}-${AWS_BATCH_JOB_ATTEMPT}"
+    # AWS_BATCH_JOB_ID is in the format job_id#node_id. Stripping #node_id.
+    shared_dir_temp="${shared_dir}/${AWS_BATCH_JOB_ID%#*}-${AWS_BATCH_JOB_ATTEMPT}"
 
     mkdir -p "${shared_dir_temp}"
 
+    write_compute_address
```

Comment:
I'd rename this function, since it is used by both compute and master nodes (E.g. `write_own_address` or `write_address`)

Reply:
Agreed!